tests/acceptance/features/apiSpaces/listSpaces.feature

```diff
@@ -19,6 +19,35 @@ Feature: List and create spaces
       | name             | Alice Hansen |
       | quota@@@state    | normal       |
       | root@@@webDavUrl | %base_url%/dav/spaces/%space_id% |
+    And the json responded should contain a space "Shares Jail" with these key and value pairs:
+      | key              | value        |
+      | driveType        | virtual     |
+      | id               | %space_id%   |
+      | name             | Shares Jail |
+      | root@@@webDavUrl | %base_url%/dav/spaces/%space_id% |
+
+  Scenario: An ordinary user can request information about their Space via the Graph API using a filter
+    When user "Alice" lists all available spaces via the GraphApi with query "$filter=driveType eq 'personal'"
+    Then the HTTP status code should be "200"
+    And the json responded should contain a space "Alice Hansen" with these key and value pairs:
+      | key              | value        |
+      | driveType        | personal     |
+      | id               | %space_id%   |
+      | name             | Alice Hansen |
+      | quota@@@state    | normal       |
+      | root@@@webDavUrl | %base_url%/dav/spaces/%space_id% |
+    And the json responded should not contain a space with name "Shares Jail"
+    And the json responded should contain spaces of type "personal"
+    And the json responded should only contain spaces of type "personal"
+    And the json responded should not contain spaces of type "project"
+    And the json responded should not contain spaces of type "virtual"
+    And the json responded should not contain spaces of type "public"
+
+  Scenario: An ordinary user will not see any space when using a filter for project
+    When user "Alice" lists all available spaces via the GraphApi with query "$filter=driveType eq 'project'"
```

Comment:
I'm not so sure here.  Is this right?  GET `https://localhost:9200/graph/v1.0/me/drives/?$filter=driveType eq 'personal'` 
I thought it might have been like: `https://localhost:9200/graph/v1.0/me/drives?driveType=personal`

Reply:
this is correct, check the odata spec https://documentation.softwareag.com/webmethods/microservices_container/msc10-5/10-5_MSC_PIE_webhelp/index.html#page/integration-server-integrated-webhelp/esb.oDataService.filter.html

Reply:
https://www.odata.org/documentation/odata-version-2-0/uri-conventions/